lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -146,14 +146,28 @@ def simple_description
 
         private
 
-        def secure_password_being_validated?
-          defined?(::ActiveModel::SecurePassword) &&
-            @subject.class.ancestors.include?(::ActiveModel::SecurePassword::InstanceMethodsOnActivation) &&
+        def should_double_check_attribute?
+          attribute_is_secure_password? ||
+            attribute_has_digest?
+        end
+
+        def attribute_is_secure_password?
+          defined?(
```

Comment:
I was reviewing Rails 6's new beta and stumbled with this issue too. I came up with the following question: shouldn't these checks be more optimistic about the attribute's name and check if it is `password` first, before doing the subsequent checks for `ActiveModel::SecurePassword`?